samples: matter: Enabled BLE extended announcement

kkasperczyk-no opened the pull request:
Enabled by default BLE extended announcement in all Matter samples. Also, enabled pairing autostart in Window Covering and Light Switch samples (according to spec all devices shall use it besides door lock, or define custom commissioning flow).

NordicBuilder commented:
## Test specification

### [CI/Jenkins/NRF](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-15371/1/)
- Integration Platforms

### [CI/Jenkins/integration](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-15371/1/)
| Test Module | File based changes | Manually selected | West overwrite |
| ----------- | :----------------: | :---------------: | :------------: |
| test-fw-nrfconnect-chip | X | | |


[Detailed information of selected test modules](https://jenkins-ncs.nordicsemi.no/job/latest/job/sdk-nrf/job/PR-15371/1/artifact/test-spec/test-spec-summary.yml/*view*/)

*Note: This message is automatically posted and updated by the CI*

NordicBuilder commented:
You can find the documentation preview for this PR at this <a href='http://developer.nordicsemi.com/nRF_Connect_SDK_dev/doc/PR-15371
'>link</a>. It will be updated about 10 minutes after the documentation build succeeds.

*Note: This comment is automatically posted by the Documentation Publishing GitHub Action.*

wiba-nordic commented on doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:
```diff
@@ -161,6 +161,19 @@ For more information about the Wi-Fi power save mechanism, see the :ref:`Wi-Fi M
+Selecting the optimal advertising duration is a compromise and depends on the specific application use case.
+To reduce the consumed energy you can consider:
+
+* Setting :kconfig:option:`CONFIG_CHIP_BLE_EXT_ADVERTISING` to ``n`` to disable Extended Announcement (also called Extended Beaconing), which allows a device to advertise for a duration of more than 15 minutes.
```
```suggestion
* :kconfig:option:`CONFIG_CHIP_BLE_EXT_ADVERTISING` -  Set to ``n`` to disable Extended Announcement (also called Extended Beaconing).
  When disabled, the device is not allowed to advertise for a duration of more than 15 minutes.
```

wiba-nordic commented on doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:
```diff
@@ -161,6 +161,19 @@ For more information about the Wi-Fi power save mechanism, see the :ref:`Wi-Fi M
+To reduce the consumed energy you can consider:
+
+* Setting :kconfig:option:`CONFIG_CHIP_BLE_EXT_ADVERTISING` to ``n`` to disable Extended Announcement (also called Extended Beaconing), which allows a device to advertise for a duration of more than 15 minutes.
+* Setting :kconfig:option:`CONFIG_CHIP_BLE_ADVERTISING_DURATION` to the lower value.
```
```suggestion
* :kconfig:option:`CONFIG_CHIP_BLE_ADVERTISING_DURATION` - Set how long the device advertises (in minutes).
```

wiba-nordic reviewed (commented):
Shouldn't we also have a changelog entry for the samples with autopairing now enabled?

kkasperczyk-no commented on doc/nrf/protocols/matter/getting_started/low_power_configuration.rst:
```diff
@@ -161,6 +161,19 @@ For more information about the Wi-Fi power save mechanism, see the :ref:`Wi-Fi M
+Extending the advertising duration may improve the user experience, as it gives more time for the user to setup the device, but it also increases the energy consumption.
+
+Selecting the optimal advertising duration is a compromise and depends on the specific application use case.
+To reduce the consumed energy you can consider:
```
I added to your suggestion "and reduce the consumed energy", because I think it's important to highlight what is the purpose of the configuration.

markaj-nordic reviewed (approved):
Should we update the documentation for Window Covering and Light Switch (BLE advertising start)?

kkasperczyk-no commented:
> Should we update the documentation for Window Covering and Light Switch (BLE advertising start)?

Fair point. @wiba-nordic please take a look once again. I updated readme files for these samples to mention that the advertising is automatically started and the duration is 1 hour. I also added entry in the release notes.

wiba-nordic commented on doc/nrf/releases_and_maturity/releases/release-notes-changelog.rst:
```diff
@@ -549,6 +549,8 @@ Matter samples
 
     DFU support for the nRF54L15 PDK is available only for the ``release`` build target.
 
+  * Enabled Bluetooth® LE Extended Announcement feature to all samples and increased advertising timeout from 15 minutes to 1 hour.
```
```suggestion
* Enabled the Bluetooth® LE Extended Announcement feature for all samples, and increased advertising timeout from 15 minutes to 1 hour.
```

wiba-nordic commented on samples/matter/light_switch/README.rst:
```diff
@@ -401,7 +401,7 @@ Commissioning the device
 Before starting the commissioning procedure, the device must be made discoverable over Bluetooth LE.
-By default, the device is not discoverable automatically upon startup.
+The device becomes discoverable automatically upon the device startup, but only for a predefined period of time (1 hour by default).
 Press **Button 1** to enable the Bluetooth LE advertising.
```
```suggestion
If the Bluetooth LE advertising times out, press **Button 1** to enable it again.
```
